src/common/reactor.cc

```diff
@@ -360,23 +346,37 @@ namespace Pistache::Aio
 
             void removeAll()
             {
+                GUARD_AND_DBG_LOG(handlers_arr_mutex_);
+                
                 index_ = 0;
                 handlers.fill(NULL);
             }
 
             std::shared_ptr<Handler> operator[](size_t index) const
             {
-                return handlers.at(index);
+                return handlers.at(index); // calls const "at"
             }
 
-            std::shared_ptr<Handler> at(size_t index) const
-            {
+            std::shared_ptr<Handler> at(size_t index)
+            { // It's const, except that the mutex is locked and unlocked
```

Comment:
@dgreatwood, if you want to guarantee to the caller that the method is logically const, you could try leaving it with a `const` qualifier, but setting the mutex as `mutable`.